pkg/core/resources/manager/cache.go

```diff
@@ -46,9 +46,13 @@ func NewCachedManager(delegate ReadOnlyResourceManager, expirationTime time.Dura
 	}, nil
 }
 
+var HashSuffix = func(ctx context.Context) string {
+	return ""
+}
```

Comment:
Can we avoid relying on globals like this. It makes things hard to test and usually wiring in the funciton in the init is easy enough

Reply:
Also instead of hasSuffix should this just be an overridable cacheKey generator?

Reply:
Or part of opts so that it's taken into account in opts.HashCode() ?

Reply:
continued in - https://github.com/kumahq/kuma/pull/6583#discussion_r1175513956 right?